:moyai: Initial Draft of Contribution Guidelines

tylermauthe opened the pull request:
I hope this is short enough to be short and sweet :+1: 

Let me know what you think, I will add some comments on the parts I am not crazy about -- but feel free to call it all :hankey:.

tylermauthe commented on CONTRIBUTING.md:
Not crazy about the word toolchain -- basically this is for Technical Story commits that don't add features or fixes.

Stuff like adding Travis CI support, ignoring files from git, etc.

tylermauthe commented on CONTRIBUTING.md:
This list could be reduced to just the emoji if we want, I find that using these guidelines improves the readability of a project's history. I've also called them suggestions, because I don't think we want to be militant about this stuff.

tylermauthe commented on CONTRIBUTING.md:
Not sure if it needs to be said, but people sometimes get really defensive when criticism is offered after they have slaved away on a PR.

tylermauthe commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,57 @@
+## Have an Issue
+If you run into problems with Dashing, please take these steps before submitting an issue:
+
+1. Check the [Troubleshooting Guide]() in the wiki.
```
I need to actually create Wiki pages for 1 and 3 in this list... :grimacing:

pushmatrix replied:
I'd just link them here for now: https://github.com/Shopify/dashing/wiki#how-tos

pushmatrix commented on CONTRIBUTING.md:
I think it's fine

pushmatrix commented on CONTRIBUTING.md:
Yeah, although it's nice to have consistency with verb tenses and moods, I don't want to make the list of "rules"/"suggestions" seem too long. I'd just stick with the emojis

pushmatrix commented on CONTRIBUTING.md:
I don't think it's necessary to mention this.

pushmatrix commented on CONTRIBUTING.md:
Awesome! That's a great idea

pushmatrix commented on CONTRIBUTING.md:
I'd leave the last line out. It's enough to say that we will discuss and critique.

tylermauthe commented on CONTRIBUTING.md:
Yeah, it provides no value -- strictly a nice to have. Removed.

tylermauthe commented on CONTRIBUTING.md:
Cool beans, I only added it because I got into a spat with someone once for critiquing his work.

tylermauthe commented on CONTRIBUTING.md:
Removed.

pushmatrix commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,57 @@
+
+1. Check the [Troubleshooting Guide]() in the wiki.
+2. Use the [GitHub Issue Search](https://help.github.com/articles/searching-issues/) to check if the issue has already been reported.
+3. Submit your issue to the [best place](). Please provide as much helpful information as possible, preferably making use of a [reduced test case](https://www.google.ca/search?#q=reduced%20test%20case).
```
Well put!

pushmatrix commented on CONTRIBUTING.md:
:(

pushmatrix commented on CONTRIBUTING.md:
Just send them this: https://vimeo.com/93947228

pushmatrix commented on CONTRIBUTING.md:
Mention that we should add them here: https://github.com/Shopify/dashing/wiki/Additional-Widgets#other-third-party-tools

pushmatrix commented on CONTRIBUTING.md:
Is there a weird newline here between idea and fits?

pushmatrix commented on CONTRIBUTING.md:
updated:
Feature requests are welcome, but take a moment to consider whether your idea fits with the scope and aim of the project. A good rule of thumb is to apply the 80/20 rule: every feature should be useful to at least 80% of users. Adding in every possible edge case will only make it more difficult to understand, maintain, and hack on.

tylermauthe commented on CONTRIBUTING.md:
More like there isn't a newline in the next line. I've gone through and broken up run-on lines -- Markdown parses it all as one `<p>` block.

pushmatrix commented:
Ah this is great! Thanks so much for doing this :)

tylermauthe commented on CONTRIBUTING.md:
Added.

tylermauthe commented on CONTRIBUTING.md:
Updated with new wording :+1:

tylermauthe commented:
Okay, I've incorporated all the feedback (I think) and I've squashed my commits.

Should we let this stew for a day or two before merging the PR or just go for it?

pushmatrix commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+for you _and_ Dashing.
+
+<a name="issues"></a>
+## Have an Issue
```
Small nitpick. A questions mark at the end of this

pushmatrix replied:
I'll edit this on master

pushmatrix commented:
Amazing! I'm merging now

tylermauthe commented:
Yeah, I like the way squash commits look because it shows the original history beneath the squash message -- but still shows up as one commit. Nice for fixing up PRs. I should have customised the squash message to be a summary of the squashed commits :+1:.

I probably didn't even need to squash these... But squashing commits is so much fun! I use SVN on a day-to-day basis so I like to git it up as much as I can when I do.
